spaceship.zsh

```diff
@@ -293,11 +293,18 @@ _is_git() {
   command git rev-parse --is-inside-work-tree &>/dev/null
 }
 
-# Check if the current direcotory is in a Mercurial repository
+# Check if the current directory is in a Mercurial repository.
 # USAGE:
-#   _is_hg
+#   if [ $(_is_hg) ]
 _is_hg() {
-  command hg --cwd $PWD root &>/dev/null
+  local root="$(pwd -P)"
+
+  while [[ $root && ! -d $root/.hg ]]
+  do
+    root="${root%/*}"
+  done
+
+  echo "$root"
```

Comment:
`$root` is just a local variable which is use to represent the current directory we're looking in. What I do is check parent directories of the current directory and try to find `.hg` directory. 

I'm not really familiar with the `function || return` syntax but it's probably easy to do. We just need to return true if `$root` is not empty and return false otherwise. How would I do that to respect the `_is_hg() || return` syntax?